### PR TITLE
add automated tests for the chart

### PR DESCRIPTION
**Pull request by QuantumEnigmaa**
Towards https://github.com/giantswarm/roadmap/issues/2008

### CODE REVIEW

**Comment by QuantumEnigmaa** [comment]
It's finally working !

File: .ats/main.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+upgrade-tests-cluster-type: kind
+upgrade-tests-cluster-config-file: tests/kind-config.yaml
+upgrade-tests-app-catalog-url: https://giantswarm.github.io/giantswarm-catalog
+upgrade-tests-app-config-file: tests/test-values.yaml
+
+skip-steps: [functional]
+
+# --------
+# Uncomment values below this line to be able to execute app-test-suite against an externally created
```
Comment (QuentinBisson) [inline .ats/main.yaml:15]: What are those comments?

File: tests/ats/test_app.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import pykube
+from pytest_helm_charts.clusters import Cluster
+from pytest_helm_charts.k8s.deployment import wait_for_deployments_to_run
+from pytest_helm_charts.k8s.stateful_set import wait_for_stateful_sets_to_run
+
+
+logger = logging.getLogger(__name__)
+
+namespace_name = "loki"
+write_sts_name = "loki-write"
```
Comment (QuentinBisson) [inline tests/ats/test_app.py:14]:
```suggestion
write_statefulset_name = "loki-write"
```

You know the drill 😅

File: tests/ats/test_app.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    Please refer to [pykube](https://pykube.readthedocs.io/en/latest/api/pykube.html) to get docs
+    for [HTTPClient](https://pykube.readthedocs.io/en/latest/api/pykube.html#pykube.http.HTTPClient).
+    """
+    assert kube_cluster.kube_client is not None
+    assert len(pykube.Node.objects(kube_cluster.kube_client)) >= 1
+
+# scope "module" means this is run only once, for the first test case requesting! It might be tricky
+# if you want to assert this multiple times
+@pytest.fixture(scope="module")
+def ic_components(kube_cluster: Cluster) -> Tuple[List[pykube.Deployment], List[pykube.StatefulSet]]:
```
Comment (QuentinBisson) [inline tests/ats/test_app.py:36]: ic is there again 😅

File: tests/test-values.yaml [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    resources:
+      limits:
+        memory: 2Gi
+      requests:
+        cpu: 100m
+        memory: 1Gi
+      policyTypes:
+        - Egress
+multiTenantAuth:
+  deployCredentials: true
```
Comment (QuentinBisson) [inline tests/test-values.yaml:216]: I would test without mtp

Comment (QuentinBisson) [reply tests/test-values.yaml:216]: But really nice work

File: tests/ats/test_app.py (hunk 3 above)
Comment (QuantumEnigmaa) [reply tests/ats/test_app.py]: I don't know what you're talking about :angel:

File: tests/ats/test_app.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+    return pods
+
+# when we start the tests on circleci, we have to wait for pods to be available, hence
+# this additional delay and retries
+
+@pytest.mark.smoke
+@pytest.mark.upgrade
+@pytest.mark.flaky(reruns=5, reruns_delay=10)
+def test_pods_available(ic_components: Tuple[List[pykube.Deployment], List[pykube.StatefulSet]]):
```
Comment (QuentinBisson) [inline tests/ats/test_app.py:75]:
```suggestion
def test_pods_available(components: Tuple[List[pykube.Deployment], List[pykube.StatefulSet]]):
```

You forgot this one but that's not important

File: .ats/main.yaml (hunk 1 above)
Comment (QuantumEnigmaa) [reply .ats/main.yaml:15]: This is an additional config for ats to enable to run against externally created kind clusters (such as a local one)

Comment (QuentinBisson) [reply .ats/main.yaml:15]: okay :)

File: tests/test-values.yaml (hunk 4 above)
Comment (QuantumEnigmaa) [reply tests/test-values.yaml:216]: Why that ? :thinking:

Comment (QuentinBisson) [reply tests/test-values.yaml:216]: Well you're not testing that it's ready right?

Comment (QuantumEnigmaa) [reply tests/test-values.yaml:216]: Well yeah, you have a point there.

File: tests/ats/test_app.py (hunk 5 above)
Comment (QuantumEnigmaa) [reply tests/ats/test_app.py]: Fixed

**Review by TheoBrigitte — APPROVED** [review]
Seems pretty good to me

**Comment by TheoBrigitte** [comment]
It would be good to have the values file somehow in sync with what we have in production.

File: tests/test-values.yaml [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+        object_store: s3
+        schema: v12
+        store: tsdb
+      - from: "2024-05-23"
+        index:
+          period: 24h
+          prefix: loki_index_
+        object_store: s3
+        schema: v13
+        store: tsdb
```
Comment (hervenicol) [inline tests/test-values.yaml:151]: We should ditch the old schemaconfigs.

```suggestion
      - from: "2024-05-23"
        index:
          period: 24h
          prefix: loki_index_
        object_store: s3
        schema: v13
        store: tsdb
```

**Review by hervenicol — COMMENTED** [review]
References to golem should be avoided.

File: tests/test-values.yaml (hunk 6 above)
Comment (QuantumEnigmaa) [reply tests/test-values.yaml:151]: Fine with me, I removed those.

**Comment by QuantumEnigmaa** [comment]
In addition to what was suggested, I also disabled the loki-multi-tenant-proxy as well as the ingress to match this : https://github.com/giantswarm/shared-configs/pull/147/files

**Comment by QuantumEnigmaa** [comment]
> It would be good to have the values file somehow in sync with what we have in production.

The test-values I use is as close as it can be from a production loki as it's basically grizzly's chart-values tuned just enough to make it work in the ATS environment.

**Pull request merged**